node/src/node_configurator/mod.rs

```diff
@@ -11,34 +11,218 @@ use crate::database::db_initializer::{DbInitializer, DbInitializerReal};
+    chain: &Chain,
+) -> (PathBuf, bool) {
+    match argument_from_vcl(configs, "--data-directory") {
+        Some(data_dir) => match PathBuf::from(data_dir).starts_with("~/") {
+            true => {
+                let home_dir_from_wrapper = dirs_wrapper
+                    .home_dir()
+                    .expect("expexted users home dir")
+                    .to_str()
+                    .expect("expect home dir")
+                    .to_string();
+                let replaced_tilde_dir =
+                    data_dir
+                        .to_string()
+                        .replacen('~', home_dir_from_wrapper.as_str(), 1);
+                (PathBuf::from(replaced_tilde_dir), true)
+            }
+            false => (PathBuf::from(&data_dir), true),
+        },
+        None => (
+            data_directory_from_context(dirs_wrapper, real_user, *chain),
+            false,
+        ),
+    }
+}
+
+fn get_config_file_from_vcl(
+    configs: &[Box<dyn VclArg>],
+    data_directory: &PathBuf,
+    data_directory_def: bool,
+    dirs_wrapper: &dyn DirsWrapper,
+) -> (PathBuf, bool) {
+    match argument_from_vcl(configs, "--config-file") {
+        Some(config_str) => {
+            let path = match PathBuf::from(config_str).is_relative() {
+                true => {
+                    match PathBuf::from(config_str).file_name().expect("expected file name") == config_str {
+                        true => PathBuf::from(data_directory).join(PathBuf::from(config_str)),
+                        false => match PathBuf::from(config_str).starts_with("./") || PathBuf::from(config_str).starts_with("../") {
+                            true => current_dir().expect("expected curerrnt dir").join(PathBuf::from(config_str)),
+                            false => match PathBuf::from(config_str).starts_with("~") {
+                                true => {
+                                    let home_dir_from_wrapper = dirs_wrapper
+                                        .home_dir()
+                                        .expect("expexted users home_dir")
+                                        .to_str()
+                                        .expect("expect str home_dir")
+                                        .to_string();
+                                    let replaced_tilde_dir =
+                                        config_str
+                                            .to_string()
+                                            .replacen('~', home_dir_from_wrapper.as_str(), 1);
+                                    PathBuf::from(replaced_tilde_dir)
+                                }
+                                false => match data_directory_def {
+                                    true => PathBuf::from(data_directory).join(PathBuf::from(config_str)),
+                                    false => panic!("You need to define data-directory to define config file with naked directory 'dirname/config.toml'.")
```

Comment:
Can you modify this error message to contain the real strings that came in instead of example strings?